kayenta-stackdriver/src/main/java/com/netflix/kayenta/stackdriver/metrics/StackdriverMetricsService.java

```diff
@@ -414,20 +414,31 @@ public String buildQuery(
               .stepMillis(alignmentPeriodSec * 1000)
               .values(pointValues);
 
-      // TODO(duftler): Still not quite sure whether this is right or if we should use
-      // timeSeries.getMetric().getLabels() instead.
-      MonitoredResource monitoredResource = timeSeries.getResource();
+      Map<String, String> filteredLabels = new HashMap<>();
 
+      MonitoredResource monitoredResource = timeSeries.getResource();
       if (monitoredResource != null) {
         Map<String, String> labels = monitoredResource.getLabels();
 
         if (labels != null) {
-          Map<String, String> prunedLabels = new HashMap<>(labels);
-          prunedLabels.remove("project_id");
-          metricSetBuilder.tags(prunedLabels);
+          filteredLabels.putAll(
+              labels.entrySet().stream()
+                  .filter(entry -> entry.getKey() != "project_id")
+                  .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)));
+        }
+      }
+
+      Metric metric = timeSeries.getMetric();
+      if (metric != null) {
+        Map<String, String> labels = metric.getLabels();
+
+        if (labels != null) {
+          filteredLabels.putAll(labels);
         }
       }
 
+      metricSetBuilder.tags(filteredLabels);
```

Comment:
Whats the difference between what labels are returned by `monitoredResource.getLabels();` vs `metric.getLabels();`?

Reply:
@fieldju `monitoredResource.getLabels();` returns the labels for the reporting GCP resource (e.g., `project_id`, `pod_id`, `namespace_id`) and are usually automatically added to custom metrics. `metric.getLabels();` returns the labels defined by the metric descriptor and are user-defined for custom metrics (e.g., `status_code`, `controller`, `action`).

Reply:
Is there a reason to not have those be separate things in the metadata?

```metricSetBuilder.attribute("metric-labels, metric.getLabels());```

Reply:
I didn't find anything in the code that was using attributes for creating comparison pairs.